app/src/components/report/ReportEdit.tsx

```diff
@@ -0,0 +1,209 @@
+import Form from 'react-bootstrap/Form';
+import Spinner from 'react-bootstrap/Spinner';
+import Stack from 'react-bootstrap/Stack';
+import ButtonLink from '../ButtonLink';
+
+export type ReportEditProps = {
+    report: Omit<IReport, 'reportId'> | undefined;
+    setReport: (newReport: IReport) => void;
+    submitLabel: string;
+    isSubmitLoading: boolean;
+    onSubmit: () => void;
+    cancelHref: string;
+};
+
+/** Report editor, used for create and edit flows */
+export default function ReportEdit({
+    report,
+    setReport,
+    submitLabel,
+    isSubmitLoading,
+    onSubmit,
+    cancelHref,
+}: ReportEditProps) {
+    const updateReport = useCallback(
+        (newValues: Partial<IReport>) => {
+            setReport({
+                ...(report as IReport),
+                ...newValues,
+            });
+        },
+        [report, setReport]
+    );
+
+    return (
+        <Form>
+            <Form.Label>Contact Information</Form.Label>
+
+            <Form.Group className="mb-3" controlId="EditReport.Name">
+                <Form.Label>Name</Form.Label>
+                <Form.Control
+                    type="text"
+                    placeholder="Please enter your name"
+                    defaultValue={report?.name}
+                    onChange={(event) => updateReport({ name: event.target.value })}
+                />
+            </Form.Group>
+
+            <Form.Group className="mb-3" controlId="EditReport.EmailAddress">
+                <Form.Label>Email Address</Form.Label>
+                <Form.Control
+                    type="email"
+                    placeholder="Please enter your email address"
+                    defaultValue={report?.emailAddress}
+                    onChange={(event) => updateReport({ emailAddress: event.target.value })}
+                />
+            </Form.Group>
+
+            <Form.Group className="mb-3" controlId="EditReport.PhoneNumber">
+                <Form.Label>Phone Number</Form.Label>
+                <Form.Control
+                    type="tel"
+                    placeholder="Please enter your phone number"
+                    defaultValue={report?.phoneNumber}
+                    onChange={(event) => updateReport({ phoneNumber: event.target.value })}
+                />
+            </Form.Group>
+
+            <Form.Group className="mb-3" controlId="EditReport.ReportCategory">
+                <Form.Label>Report Category</Form.Label>
+                {[
+                    'Illegal Dumping',
+                    'Clogged Storm Drain',
+                    'Potholes',
+                    'Graffiti',
+                    'Street Light Outage',
+                    'Sidewalk Damage',
+                    'Traffic Signal Malfunction',
+                    'Abandoned Vehicles',
+                    'Noise Complaint',
+                    'Other',
+                ].map((category) => (
+                    <Form.Check
+                        type="checkbox"
```

Comment:
Is there a reason for each report to have multiple categories? I think it would make sense for each report to have a single category so that city employees can appropriately route the ticket to a single department